### PR TITLE
Add tests for AudienceQuery objects and JSON parsing

### PR DESCRIPTION
**Pull request by dvoineu**

### CODE REVIEW

File: EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvPredicateImpl.swift [hunk 1]
```diff
@@ -21,9 +21,54 @@ import Foundation
...
+        
+        static func defined (a: String?) -> Bool {
+            return a != nil
+        }
+        
+        static func equal<T: Equatable> (a: T, b: T) -> Bool {
+            return a == b
+        }
+        
+        static func exists (a: String) -> Bool {
```
Comment (robertsevernsentient) [inline EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvPredicateImpl.swift:39]: This should be verifying the the key exists in the context - returning true feels incorrect here

File: EvolvSwiftSDK/EvolvSwiftSDK/Sources/Utils/OrderedDictionary.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
+//
```
Comment (robertsevernsentient) [inline EvolvSwiftSDK/EvolvSwiftSDK/Sources/Utils/OrderedDictionary.swift:1]: Can you add tests for this file?

**Review by robertsevernsentient — CHANGES_REQUESTED** [review]
Requested tests for the ordered dictionary and an update to the exists evaluation

File: EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvPredicateImpl.swift (hunk 1 above)
Comment (dvoineu) [reply EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvPredicateImpl.swift:39]: Exactly. Forgot to change the implementation

File: EvolvSwiftSDK/EvolvSwiftSDK/Sources/Utils/OrderedDictionary.swift (hunk 2 above)
Comment (dvoineu) [reply EvolvSwiftSDK/EvolvSwiftSDK/Sources/Utils/OrderedDictionary.swift:1]: yes. I will add them before our next meeting

**Review by robertsevernsentient — APPROVED** [review]

**Pull request merged**